src/core/plugins

```diff
@@ -0,0 +1,306 @@
+    echo "Plugins ${list_status}: ${plugins%,}"
+    log INFO "Plugins ${list_status}: ${plugins%,}"
+  fi
+  return 0
+}
+
+# Validate plugin names. "all" is valid (represent all available)
+validate_plugin() {
+  local plugins="$1"
+  local -a plugins_a
+
+  # Validate "all", it is not valid to combine "all" with other plugins
+  if [[ "${plugins}" == "all" ]]; then
+    return 0
+  fi
+  
+  # Getting the list of available plugins
+  plugins_a=( $( get_plugin_status "available" ) )
+
+  # Create array with the provided plugins, remove dups
+  local -a p_v=( $( csv_remove_dups "${plugins}" ) )
+  
+  # Validate provided plugin with available
+  for p in ${p_v[@]}; do
+    if ! in_array ${p} ${plugins_a[@]}; then
+        return 1
+    fi
+  done
+
+  return 0
+}
+
+# Enable Plugin support in recap.conf
+enable_plugin_conf() {
+   local recap_conf="/etc/recap.conf"
+
+   # Only perform changes when not already defined
+   if [[ "${USEPLUGINS}" == "no" ]]; then
+     # Comment out explicitly disabled
+     if grep -q '^USEPLUGINS="no"' "${recap_conf}"; then
+        sed -i \
+          -e 's/^\(USEPLUGINS="no"*\)$/#\1/' \
+          "${recap_conf}"
+     fi
+     # Comment out all explicitly enabled, we enable them next
+     # This is to remove dups
+     if grep -q '^USEPLUGINS="yes"' "${recap_conf}"; then
+        sed -i \
+          -e 's/^\(USEPLUGINS="yes"*\)$/#\1/' \
+          "${recap_conf}"
+     fi
+
+     # Un-comment out if enabling is commented out
+     if grep -P -q '^#+USEPLUGINS="yes"' "${recap_conf}"; then
+        sed -E -i \
+          -e '0,/^#+USEPLUGINS="yes"/s//USEPLUGINS="yes"/' \
+          "${recap_conf}"
+
+     # Add after the first reference commented out
+     elif grep -P -q '^#+USEPLUGINS="no"' "${recap_conf}"; then
+        sed -E -i \
+          -e '0,/^#+USEPLUGINS="no"/{
+	       /^#+USEPLUGINS="no"/a USEPLUGINS="yes"
+	  }' "${recap_conf}"
+
+     # Add at the end, as no traces have been found
+     # Using sed, just to keep it consistent
+     else
+        sed -i \
+	  -e '$ a\#USEPLUGINS="no"\nUSEPLUGINS="yes"' \
+          "${recap_conf}"
+     fi
+     log INFO "Plugins enabled in ${recap_conf}"
+   fi
+}
+
+# Plugin Enable/Disable
+plugin_action() {
+  local action="${1}"
+  local plugins="${2}"
+
+  # Get the other (opposite) action
+  local -a other_action=( "enable" "disable" )
+  other_action=( ${other_action[@]//${action}} )
+
+  # Get plugins (disabled or enabled)
+  local p_a=( $( get_plugin_status "${other_action}d" ) )
+
+  # Cleanup provided plugins or get "all" plugins
+  local -a p_2a
+  local -a actioned
+  if [[ "${plugins}" == "all" ]]; then
+    p_2a=( $( get_plugin_status "available" ) )
+  else
+    p_2a=( $( csv_remove_dups "${plugins}" ) )
+  fi
+
+  # Enable or disable plugins that are currently disabled or enabled
+  for plugin in ${p_2a[@]}; do
+    if in_array ${plugin} ${p_a[@]}; then
+      local symlink="${PLUGIN_E_DIR}/${plugin}"
+
+      # Enable disabled plugin
```

Comment:
I think we could increase readability by abstracting the actual filesystem operations within "enable_plugin" and "disable_plugin" helper functions.